libpod/runtime.go

```diff
@@ -0,0 +1,75 @@
+package libpod
+
+import (
+	"fmt"
+
+	"github.com/kubernetes-incubator/cri-o/libpod/ctr"
+	"github.com/kubernetes-incubator/cri-o/libpod/pod"
+	spec "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+var (
+	// ErrNotImplemented indicates that functionality is not yet implemented
+	ErrNotImplemented = fmt.Errorf("NOT IMPLEMENTED")
+)
+
+// Runtime is the core libpod runtime
+type Runtime struct {
+	// TODO populate
+}
+
+// NewContainer creates a new container from a given OCI config
+func (r *Runtime) NewContainer(spec *spec.Spec, options ...func(*ctr.Container) error) (*ctr.Container, error) {
+	return nil, ErrNotImplemented
+}
+
+// RemoveContainer removes the given container
+// If force is specified, the container will be stopped first
+// Otherwise, RemoveContainer will return an error if the container is running
+func (r *Runtime) RemoveContainer(ctr *ctr.Container, force bool) error {
+	return ErrNotImplemented
+}
+
+// GetContainer retrieves a container by its ID
+func (r *Runtime) GetContainer(id string) (*ctr.Container, error) {
+	return nil, ErrNotImplemented
+}
+
+// LookupContainer looks up a container by its name or a partial ID
+// If a partial ID is not unique, an error will be returned
+func (r *Runtime) LookupContainer(idOrName string) (*ctr.Container, error) {
+	return nil, ErrNotImplemented
+}
+
+// GetAllContainers retrieves all containers from the state
+func (r *Runtime) GetAllContainers() ([]*ctr.Container, error) {
+	return nil, ErrNotImplemented
+}
+
+// NewPod makes a new pod
+func (r *Runtime) NewPod() (*pod.Pod, error) {
+	return nil, ErrNotImplemented
+}
+
+// RemovePod removes a pod and all containers in it
+// If force is specified, all containers in the pod will be stopped first
+// Otherwise, RemovePod will return an error if any container in the pod is running
+func (r *Runtime) RemovePod(pod *pod.Pod, force bool) error {
+	return nil, ErrNotImplemented
+}
+
```

Comment:
That works.  Might be nice to bump up the comments then a bit to explain that. FWIW, it still might be useful to have the SafeToRemovePod function to keep the removepod function smaller and it might be useful elsewhere.  Obviously you'd want to call that inside the lock.  Renaming the suggested function to "AreContainersInPodStopped" (or something more creative) would be OK too and it might make more sense named that way in other spots. 

Anywho, my $.02, unfortunately I can't even buy you a cup of coffee with that these days.